src/core/markdown.ts

```diff
@@ -8,6 +8,7 @@ import { preprocessHead } from './head'
 
 const scriptSetupRE = /<\s*script([^>]*)\bsetup\b([^>]*)>([\s\S]*)<\/script>/mg
 const defineExposeRE = /defineExpose\s*\(/mg
+const internalLinkRE = /\[(.+)\]\((\/[?=\w\/-]*)\)/g
```

Comment:
I am worried that this regex can lead to some false passive. Could we do it with a markdown it plugin?

Reply:
Sure. Any you recommend? If not, I'll look around for a plugin.

Reply:
You're right about this regex. It captured images `![my-image](/examaple)` and it isn't correctly capturing links that are in the same line. I will flesh it out some more and push to this PR. Haven't found an obvious plugin to use just yet.

Reply:
@f3ve Ref: https://github.com/vuepress/core/blob/7b9cfdc416ed395d77c206e4f4bf787d218983de/packages/markdown/src/plugins/linksPlugin/linksPlugin.ts#L122-L126

Reply:
I'm not sure if this is a common feature. Both vuepress and vitepress have lots of different customizations in their links plugin.

If it could be extract to a more customizable markdown-it plugin, maybe we can add it to https://github.com/mdit-vue/mdit-vue

Reply:
Oh I see! I misunderstood what Antfu meant by we could do it in a plugin. This makes sense and thanks for the references! I'll look into developing this into a markdown-it plugin instead. Thanks so much for both of your time.